homeassistant/components/esphome/manager.py

```diff
@@ -280,42 +280,44 @@ async def _send_home_assistant_state(
 
         await self.cli.send_home_assistant_state(entity_id, attribute, str(send_state))
 
+    async def _send_home_assistant_state_event(
+        self,
+        attribute: str | None,
+        event: EventType[EventStateChangedData],
+    ) -> None:
+        """Forward Home Assistant states updates to ESPHome."""
+        event_data = event.data
+        new_state = event_data["new_state"]
+        old_state = event_data["old_state"]
+
+        if new_state is None or old_state is None:
+            return
+
+        # Only communicate changes to the state or attribute tracked
+        if (not attribute and old_state.state == new_state.state) or (
+            attribute
+            and old_state.attributes.get(attribute)
+            == new_state.attributes.get(attribute)
+        ):
+            return
+
+        await self._send_home_assistant_state(
+            event.data["entity_id"], attribute, new_state
+        )
```

Comment:
This doesn't actually await anything so it could be changed to be a normal function